x-pack/test/performance/tests/playwright/ecommerce_dashboard.ts

```diff
@@ -5,52 +5,68 @@
+                protocol: config.get('servers.kibana.protocol'),
+                hostname: config.get('servers.kibana.hostname'),
+                port: config.get('servers.kibana.port'),
+              });
 
-    step('Go to Sample Data Page', async ({ page }) => {
-      const kibanaUrl = Url.format({
-        protocol: config.get('servers.kibana.protocol'),
-        hostname: config.get('servers.kibana.hostname'),
-        port: config.get('servers.kibana.port'),
-      });
+              await page.goto(`${kibanaUrl}/app/home#/tutorial_directory/sampleData`);
+              await page.waitForSelector('text="More ways to add data"');
+            },
+          },
+          {
+            name: 'Add Ecommerce Sample Data',
+            handler: async ({ page }: { page: Page }) => {
+              const removeButton = page.locator('[data-test-subj=removeSampleDataSetecommerce]');
+              try {
+                await removeButton.click({ timeout: 1_000 });
+              } catch (e) {
+                logger.info('Ecommerce data does not exist');
+              }
+              const addDataButton = page.locator('[data-test-subj=addSampleDataSetecommerce]');
+              if (addDataButton) {
+                await addDataButton.click();
+              }
+            },
+          },
+          {
+            name: 'Go to Ecommerce Dashboard',
+            handler: async ({ page }: { page: Page }) => {
+              await page.click('[data-test-subj=launchSampleDataSetecommerce]');
+              await page.click('[data-test-subj=viewSampleDataSetecommerce-dashboard]');
 
-      await page.goto(`${kibanaUrl}/app/home#/tutorial_directory/sampleData`);
-      await page.waitForSelector('text="More ways to add data"');
-    });
-
-    step('Add Ecommerce Sample Data', async ({ page }) => {
-      const removeButton = page.locator('[data-test-subj=removeSampleDataSetecommerce]');
-      try {
-        await removeButton.click({ timeout: 1_000 });
-      } catch (e) {
-        logger.info('Ecommerce data does not exist');
-      }
-      const addDataButton = page.locator('[data-test-subj=addSampleDataSetecommerce]');
-      if (addDataButton) {
-        await addDataButton.click();
-      }
-    });
-
-    step('Go to Ecommerce Dashboard', async ({ page }) => {
-      await page.click('[data-test-subj=launchSampleDataSetecommerce]');
-      await page.click('[data-test-subj=viewSampleDataSetecommerce-dashboard]');
-
-      await page.waitForFunction(() => {
-        const visualizations = Array.from(document.querySelectorAll('[data-rendering-count]'));
-        const visualizationElementsLoaded = visualizations.length > 0;
-        const visualizationAnimationsFinished = visualizations.every(
-          (e) => e.getAttribute('data-render-complete') === 'true'
-        );
-        return visualizationElementsLoaded && visualizationAnimationsFinished;
-      });
+              await page.waitForFunction(() => {
```

Comment:
nit: 
```suggestion
              await page.waitForFunction(function renderCompeted() {
```
here and below